src/components/Toolbar/Index.tsx

```diff
@@ -0,0 +1,92 @@
+import {
+    StyleSheet,
+    ToolbarAndroid,
+    ToolbarAndroidAction
+} from 'react-native';
+import * as React from 'react';
+import icons, {
+    images
+} from '../../assets/index';
+
+/*Only one button is permitted on the left
+it must either be a menu button which has a 'burger' icon
+or a back button which has an arrow icon*/
+export enum LeftButtonTypes {
+    menu = "menu",
+    back = "back"
+}
+/*Multiple buttons can be added to the right side of the toolbar
+if an icon isn't specified the title will be displayed in its place*/
+interface RightButtonType {
```

Comment:
Is there a way we could also make the title optional? For example we might only want an icon button

Reply:
I had considered this, although the nice thing about forcing a title on the buttons is that all buttons will have some sort of description otherwise they might get lazy and only add an icon which could make the code hard to read.

Reply:
I actually agree with this too, although I think that if a title and an icon are provided, only the icon should be shown. Even for logging purposes this is probably good. If we are going to have this strict rule that icons should have a title, maybe having an `Icon` interface where:
```
interface Icon {
  image: <whatever type this is>;
  title: string;
}
```

And export these from assets instead. Unless you think the title is going to change depending on the usage, then this interface could be in Toolbar :D

Reply:
by default it only shows the icon if both are specified. You have to set a flag to True if you want both to show (speaking of which I don't have that functionality implemented)